### DISCUSSION TITLE
The non-www version of my website doesn't work

### DISCUSSION THREAD

**Discussion by majdhajyehia** [vercel/community#7428] [Help]
### Summary

I have a website setup using the hobby plan and I have it working just fine when using the www version of the domain, the non-www gives me a "took too long to respond" error message, I have already setup a redirect to www version from the domain tab in the settings and gave it enough time to propagate but without any success, it sometimes works and sometimes doesn't (recently it's not working at all).

### Steps to Reproduce

1. Setup a vanilla JS website using the hobby plan.
2. Connect a custom domain.
3. Do all recommended steps regarding DNS settings.
4. Access the website and it doesn't work!!!

**Comment by BorhanSaflo** [comment]
It sounds like this could be a DNS configuration problem. Can you confirm that the A record of your custom domain is pointing to the IP address provided by Vercel?

You can use a [DNS Checker](https://dnschecker.org) tool to see what your custom domain is pointing to and ensure it's correctly configured and propagated.

**Reply by majdhajyehia** [reply]
![image](https://github.com/vercel/community/assets/33839477/6a91c30b-cc76-4356-acbc-4b52cb641091)
aren't these configurations enough for it to work though? this is hugely frustrating, I've been struggling with this for an entire week and the vercel team basically has no way to get in touch with, I'd highly appreciate it if you could go through these settings and tell me what's wrong/missing.
Thanks in advance for the help.

**Reply by BorhanSaflo** [reply]
Your configuration seems to be correct. Use a [DNS Checker](https://dnschecker.org) to confirm that it has propagated and is resolving to that IP address. If it is resolving correctly, the issue might be related to your DNS cache or server. First, try clearing your DNS cache. If that doesn't solve the problem, try changing your DNS server. Here's a [guide](https://lazyadmin.nl/it/fix-this-site-cant-be-reached) for Windows.

**Reply by majdhajyehia** [reply]
I've tried flushing the DNS cache, I've tried deleting cookies, DNS config in chrome, reset all settings. I checked if it has propagated and it did.
But if I do change my DNS server and it works doesn't this mean all people will need to do the same? it's really weird to me that it's not working when the www version is working just fine.

**Reply by majdhajyehia** [reply]
![image](https://github.com/vercel/community/assets/33839477/a42d3247-d176-4def-94cc-ef6700b05b51)
this is the error I'm getting, do you think it's related to DNS? I think it might be taking a lot of time to resolve but I already tried everything really :(
do you have any suggestions?

**Reply by BorhanSaflo** [reply]
> But if I do change my DNS server and it works doesn't this mean all people will need to do the same?

No, not everyone needs to do the same. When you change your DNS resolver, you're just changing which DNS server your computer queries to find IP addresses. As you saw in the DNS checker, there are thousands of DNS servers around the world. Most people use their ISP's DNS resolver by default, but some use Google’s, Cloudflare’s, or other resolvers (The guide that I've linked above is using Google's).

The DNS resolver you're currently using (most likely your ISP's) might be experiencing temporary issues, which others may not encounter. That's why I suggested trying a different DNS resolver.

Here's a simplified explanation of how DNS resolution works:

1. **Your Computer:** "What's the IP address for `mywebsite.com`?"
2. **Local Cache:** "I don't have it stored, let's ask the DNS resolver."
3. **DNS Resolver:** "Root server, where is the `.com` TLD server?" 
5. **Root Server:** "Here is the address for the `.com` TLD server."
6. **DNS Resolver:** "TLD server, where is the authoritative server for `mywebsite.com`?"
7. **TLD Server:** "Here is the address for the authoritative server for `mywebsite.com`."
8. **DNS Resolver:** "Authoritative server, what is the IP address for `mywebsite.com`?"
9. **Authoritative Server:** "The IP address for `mywebsite.com` is `76.76.21.21`." (This is what you've configured earlier)
10. **DNS Resolver:** "Here is the IP address for `mywebsite.com`: `76.76.21.21`." (In this case, it's Vercel's IP)
11. **Your Computer:** "Great! Now I can visit `mywebsite.com`."

Here's a diagram:

![image](https://github.com/vercel/community/assets/60056206/8895b2ac-fb60-472a-a03e-bd9357c1024f)

I hope this helps!

[a comment by an automated account is not shown]
